internal/cli/kraft/compose/logs/logs.go

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2024, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file except in compliance with the License.
+
+package logs
+
+import (
+	"context"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/compose-spec/compose-go/types"
+	"github.com/spf13/cobra"
+
+	"kraftkit.sh/cmdfactory"
+	"kraftkit.sh/compose"
+	"kraftkit.sh/log"
+
+	kernellogs "kraftkit.sh/internal/cli/kraft/logs"
+
+	"kraftkit.sh/packmanager"
+)
+
+type LogsOptions struct {
+	Follow bool `long:"follow" usage:"Follow log output"`
+
+	composefile string
+}
+
+func NewCmd() *cobra.Command {
+	cmd, err := cmdfactory.New(&LogsOptions{}, cobra.Command{
+		Short: "Print the logs of services in a project",
+		Use:   "logs",
+		Annotations: map[string]string{
+			cmdfactory.AnnotationHelpGroup: "compose",
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	return cmd
+}
+
+func (opts *LogsOptions) Pre(cmd *cobra.Command, _ []string) error {
+	ctx, err := packmanager.WithDefaultUmbrellaManagerInContext(cmd.Context())
+	if err != nil {
+		return err
+	}
+
+	cmd.SetContext(ctx)
+
+	if cmd.Flag("file").Changed {
+		opts.composefile = cmd.Flag("file").Value.String()
+	}
+
+	log.G(cmd.Context()).WithField("composefile", opts.composefile).Debug("using")
+
+	return nil
+}
+
+func (opts *LogsOptions) Run(ctx context.Context, args []string) error {
+	workdir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	project, err := compose.NewProjectFromComposeFile(ctx, workdir, opts.composefile)
+	if err != nil {
+		return err
+	}
+
+	if err := project.Validate(ctx); err != nil {
+		return err
+	}
+
+	return opts.PrintLogs(ctx, project)
+}
+
+func (opts *LogsOptions) PrintLogs(ctx context.Context, project *compose.Project) error {
+	var wg sync.WaitGroup
```

Comment:
shouldn't this be private? Is it used anywhere else?

`printLogs`

Reply:
Nvm, saw it used  in some other place also below.

Move it please to a `internal/cli/kraft/compose/utils/utils.go` file and make public there